source_gen/README.md

```diff
@@ -117,6 +117,33 @@ targets:
 If you provide a builder that uses `SharedPartBuilder` and `combining_builder`,
 you should document this feature for your users.
 
+### Generating files in different directories
+
+When using shared-part builders which apply the `combining_builder` as part of
+the build, the output location for an input file can be changed.
+By default, a `.g.dart` file next to the input is generated.
+
+To change this, set the `build_extensions` option on the combining builder. In
```

Comment:
See https://github.com/dart-lang/build/issues/3172 - we should tell people to add a dev dependency on source_gen and require at least the version this feature ships in.